File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02912/p02912_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    
    sort(A.begin(), A.end(), greater<int>());
    
    int total = 0;
    int discount = 0;
    for (int i = 0; i < N; i++) {
        int price = A[i];
        int used = min(M, discount);
        total += price / (1 << used);
        
        if (used < M) {
            discount++;
        }
    }
    
    cout << total << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02912/p02912_num4_parsed.py
# Input
N, M = map(int, input().split())
A = list(map(int, input().split()))

# Sort the prices in decreasing order
A.sort(reverse=True)

# Initialize the total cost
total_cost = sum(A)

# Use the discount tickets
for i in range(M):
    # Divide the price by 2 and round down to the nearest integer
    A[0] = A[0] // 2

    # Sort the prices again
    A.sort(reverse=True)

# Calculate the minimum amount of money required to buy all the items
min_cost = sum(A)

# Output the result
print(min_cost)